rezz.lic

```diff
@@ -0,0 +1,184 @@
+    color = Regexp.last_match(1)
+    echo "Memories are #{color}"
+    waitrt?
+    return false unless color.eql? 'silver'
+    true
+  end
+
+  def rejuv(person)
+    data = { 'abbrev' => 'rejuv' }
+    if @rejuv_mana == 0
+      check_discern(data)
+      @rejuv_mana = data['mana']
+    end
+
+    loop do
+      cast_spell('rejuv', @rejuv_mana, person)
+      break if is_silver?(person)
+    end
+  end
+
+  def check_buffs
+    wait_for_script_to_complete('buff', ['rezz'])
+  end
+
+  def skip_harness?
+    attunement = DRSkill.getrank('Attunement')
+    if attunement < 541
+      return false
+    elsif attunement > 600
+      return true
+      elseif check_if_buff_active('Persistence of Mana', 6)
+      return true
+    else
+      false
+    end
+  end
+
+  def cast_rezz
+    data = { 'abbrev' => 'rezz' }
+    check_discern(data)
+    cast_spell('rezz', data['mana'], 'self')
+    data['mana']
+  end
+
+  def cast_soulbonding(person)
+    cast_spell('sb', 1, person, true)
+  end
+
+  def check_if_buff_active(spell, time)
+    DRSpells.active_spells[spell] && DRSpells.active_spells[spell] > time
+  end
+
+  # copied from crossing-trianing
+  def check_discern(data)
+    UserVars.discerns = {} unless UserVars.discerns
+    discern_data = UserVars.discerns[data['abbrev']] || {}
+    if data['symbiosis']
+      if discern_data.empty? || discern_data['min'].nil?
+        /requires at minimum (\d+) mana streams/ =~ bput("discern #{data['abbrev']}", 'requires at minimum \d+ mana streams')
+        discern_data['mana'] = Regexp.last_match(1).to_i
+        discern_data['cambrinth'] = nil
+        discern_data['min'] = Regexp.last_match(1).to_i
+        discern_data['more'] = 0
+      end
+      calculate_mana(discern_data['min'], discern_data['more'], discern_data, false)
+    elsif discern_data.empty? || discern_data['time_stamp'].nil? || Time.now - discern_data['time_stamp'] > 24 * 60 * 60 || !discern_data['more'].nil?
+      discern_data['time_stamp'] = Time.now
+      case discern = bput("discern #{data['abbrev']}", 'The spell requires at minimum \d+ mana streams and you think you can reinforce it with \d+ more', 'You don\'t think you are able to cast this spell', 'You have no idea how to cast that spell')
+      when /you don't think you are able/i, 'You have no idea how to cast that spell'
+        discern_data['mana'] = 1
+        discern_data['cambrinth'] = nil
+      else
+        discern =~ /minimum (\d+) mana streams and you think you can reinforce it with (\d+) more/i
+        calculate_mana(Regexp.last_match(1).to_i, Regexp.last_match(2).to_i, discern_data, data['cyclic'])
+      end
+    end
+    pause 1
+    waitrt?
+    UserVars.discerns[data['abbrev']] = discern_data
+    data['mana'] = discern_data['mana']
+    data['cambrinth'] = discern_data['cambrinth']
+  end
+
+  def calculate_mana(min, more, discern_data, cyclic)
```

Comment:
Is this copied from `crossing-training.lic`? (which has another copy in `gate.lic`)

Reply:
I have no problem with this. If I recall correctly there was an issue with doing this. I think it was a business rule. Something about yaml settings in common libs. I can move this to common-arcana if that is okay.

Reply:
i made the changes. going to test a little. Need to find a dead person though lol.

Reply:
Ah, you're right. It's the `prep_scaling_factor` and `cambrinth_cap` settings. I think we had much the same conversation in #2099 not too long ago.

Let's leave the copy/paste in for _this_ PR, but I would very much like to see the next one be extracting that logic. The catch would be that those two settings have to be passed in as parameters, not pulled out of settings like they are today.

Reply:
Im still going to test adding to common-arcana. I followed the same methodology in other common-arcana functions that require settings but providing it as an argument to the function. So you can call check_discern(data, settings) now which in turn will pass to calculate_mana.

Reply:
Took care of the commonizing of discern, the signature is as you mention
`data = check_discern(data, settings)`